### DISCUSSION TITLE
SlackAlarm을 위한 DTO 생성 vs. Interview 도메인 객체 사용

### DISCUSSION THREAD

**Discussion by HyeonbinSa** [woowacourse-teams/2022-ternoko#329] [Ideas]
기존 Controller에서 Slack 알람 기능을 사용하다 이벤트를 사용하여 의존성을 분리하였습니다. 추가로 의존성을 분리하면서 InterviewService 내 메서드를 호출하면서 알림을 사용하도록 수정해보았습니다.

### 기존
- 사용 위치 : Controller
- 사용 인자 : AlarmResponse(알람 기능을 위한 DTO)

### 현재
- 사용 위치 : InterviewService(Event로 사용)
- 사용 인자 : 각 이벤트 객체 내에 Interview 객체를 사용

### 논의 주제
- 코드 리뷰 [링크](https://github.com/woowacourse-teams/2022-ternoko/pull/325#discussion_r962881503)
- 내용 : SlackAlarm 클래스에서 Interview를 모르도록 구현할 수 있을까?

### 논의 사항
1. SlackAlarm을 위한 DTO 클래스를 만들어 사용한다.
    - 문제 : 기존 방식 AlarmResponse를 사용하여 슬랙 알람기능을 사용했을 때 SlackAlarm을 위한 DTO를 만들어야하는가라는 의견이 있었음.
    - 장점 : Interview라는 객체와의 의존성을 끊어 Interview가 아니더라도 알람을 사용할 수 있다.
2. 도메인 객체를 그대로 사용한다.
    - 문제 : 현재 알림이 Interview 객체에 한해 사용 가능하다.
    - 장정 : DTO 생성 없이 사용할 수 있다. -> Interview 객체의 수정이 생기더라도 관리에 용이함.

위와 같은 주제로 이야기할 수 있을 것 같은데 다른 분들의 의견이 궁금합니다.

**Comment by soominsohn** [comment]
저는 1번에 한표입니다!
다음과 같은 고민을 했어요.
1. SlackAlarm을 위한 DTO 클래스를 만들어 사용한다.
- SlackAlarm에서 Interview 객체의 변화가 일어나면 어떤 사이드 이펙트가 생길까?

2. 도메인 객체를 그대로 사용한다.
- 알림이 Interview객체가 아니라 다른 객체를 받을 일이 필요한가?
- 다른 로직에서 위와 비슷한 경우 dto를 반환하고 있는가?
- SlackAlarm에서 Interview 객체의 조회 로직이 아닌 다른 로직을 수행하고 있는가?

라는 고민을 했는데요, 아무래도 안정성 측면에서 `적용할 수 있는건 모두 적용해보는 경험을 해보자!` 라는 관점에서
1번 DTO 클래스를 만들어 사용한다에 한표 하겠습니다 😄 
Discussion 올려줘서 고마워요 바니!
